packages/graphiql-react/src/editor/hooks.ts

```diff
@@ -204,14 +204,23 @@ export function useMergeQuery({ caller }: UseMergeQueryArgs = {}) {
   }, [queryEditor, schema]);
 }
 
-type UsePrettifyEditorsArgs = {
+export type UsePrettifyEditorsArgs = {
   /**
    * This is only meant to be used internally in `@graphiql/react`.
    */
   caller?: Function;
+  /**
+   * Invoked when the prettify callback is invoked
+   * @param query The current value of the query editor.
+   * @returns {string} The formatted query
+   */
+  onPrettifyQuery?: (query: string) => string;
```

Comment:
```suggestion
  onPrettifyQuery?: async (query: string) => string;
```
and the required changes as such, as `prettier.format` returns a promise, for example